### DISCUSSION TITLE
英语试听刷不了 BookType为1 的这一类的教材都有问题

### DISCUSSION THREAD

**Discussion by Big-Hing** [Mufanc/iSmartAuto#27] [Q&A]
![)6ZW_ VX55D`T KE%J%M(AW](https://user-images.githubusercontent.com/106815701/171850518-84be26ef-8c91-4e86-be29-2d94dd9161cb.png)

booktype ：1 显示
2022-05-30 02:05:22.099 | INFO | automaton.utils:dfs:78 - [刷任务点] | . Unit 1 Knowledge is Great
2022-05-30 02:05:22.100 | INFO | automaton.utils:dfs:78 - [刷任务点] | . Lead-in
2022-05-30 02:05:22.101 | INFO | automaton.utils:dfs:78 - [刷任务点] | . Lead-in
......中间省略一大段相同的......
2022-05-30 02:05:22.165 | INFO | automaton.utils:dfs:78 - [刷任务点] | . Video Studio
2022-05-30 02:05:22.165 | INFO | automaton.utils:dfs:78 - [刷任务点] | . Unit 6 All on the Mind
并未提交分数，遍历所有任务目录

**Comment by beture4** [comment]
同样是booktype1 一样的情况

**Reply by Dearyou50** [reply]
一样，不知道为什么

**Reply by Javazzx** [reply]
为什么啊

**Reply by fangsancs** [reply]
作者的代码应该是是根据每道题的“paperId”属性到网页上(（https://xot-api.ismartlearning.cn/client/textbook/paperinfo）)寻找
答案，而“booktype1”的书的题目没有“paperId”这个属性，导致以下代码没有执行：（这一串代码的”if 'paperId' in task“结果为假
，代码没有执行。）
            if 'paperId' in task: # 如果有任务则提交
                chapter_id = task['chapterId']
                task_id = task['id']
                score, time = await _random(spider, task['paperId'])
                result = await page.submit(book_id, chapter_id, task_id, score, time, 100, user_id)
                if result['wasThrown'] or not result['result']['value']:
                    logger.warning(f'[刷任务点] | 任务[{task["name"]}]可能提交失败，请留意最终结果')

基于以上猜想，我尝试使用”booktype1“的书的题目的”id“属性来作为判断条件，虽然”if 'paperId' in task“结果为真，但这样以后“
获取任务点出错”，可能ismart的题目的答案是基于“paperId”属性一 一对应的。但是我猜想，”booktype1“的书可能是根据题目的
”id“属性来一 一对应答案，我猜想现在的代码需要修改题目的答案所在的网址，脚本应该就能正常运行。但是，我不会抓包，也
不懂得如何获得答案所在的网址，在座各位有能助我一臂之力的高手吗？（作者大大怎么获得答案的网址的，可以教教吗；能区
分不同题目的也只有“id”属性了，但是用这个属性去作者给的答案网址去找却会报错，所以答案网址不止一个是我的猜想；我也
研究了很久，最后一关是我的盲区了QAQ）

**Reply by fangsancs** [reply]
求作者邮箱

**Reply by puxuhui1123** [reply]
请问问题解决了吗
